### PR TITLE
Add template variables docs and warning about cookiecutter command

### LINKED ISSUE #86: Add cookiecutter variables description
### What's needed?

When creating a new project using cookiecutter, it is not obvious at all what some variables are used for, and what should be the input format, etc.

### Proposed solution

Ideally show a help message before the variables are prompted (it might be doable via the pre build hook), but also show this information in the documentation website. We can create one markdown formatted file with this information, and display it in the hook and [embed](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/#snippets) the same file (as a [snippet](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/)) in the documentation so there is only one source of truth.

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### LINKED ISSUE #93: Improve the instructions to generate a project using cookiecutter
### What's needed?

Many users reported problems while trying to generate a new project. The main mistakes were trying to modify the cookiecutter command, which should be typed literally: `cookiecutter gh:frequenz-floss/frequenz-repo-config-python --directory=cookiecutter`.

- Some users tried to put the name of their project in `gh:frequenz-floss/my-project`
- Other users wanted to put the directory they wanted generate in `--directory=my-project`

### Proposed solution

We should make it more clear that this command needs to be typed as is, and all information about the project to be generated will be prompted.

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by llucax**
- Add cookiecutter template variables description (fixes #86)
- Add a warning about the cookiecutter command (fixes #93)

### CODE REVIEW

File: cookiecutter/variable-reference.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+* `github_repo_name`: The handle of the GitHub repository where the project
+  will reside. This will be used to generate links to the project on GitHub and
+  as the top-level directory name.
+
+* `default_codeowners`: A space-separated list of GitHub teams (`@org/team`) or
+  users (`@user`) that will be the default code owners for this project. This
+  will be used to build the `CODEOWNERS` file. Please refer to the [code owners
+  documentation] for more details on the valid syntax.
+
```
Comment (Marenz) [inline cookiecutter/variable-reference.md:49]: To say it with ChatGPTs words:

> Improvements: There aren't many improvements to suggest, as the document is already well-written and detailed.

File: cookiecutter/variable-reference.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+[code owners documentation]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
```
Comment (Marenz) [inline cookiecutter/variable-reference.md:50]: Though I do think this is not a valid md syntax?

File: docs/index.md [hunk 3]
```diff
@@ -31,16 +31,29 @@ Collecting cookiecutter
...
+
+    This command needs to be typed literally!
+
+    `frequenz-floss/frequenz-repo-config-python` is the GitHub repository with
+    the cookiecutter template that will be downloaded, and
+    `--directory=cookiecutter` is needed because the cookiecutter template
+    doesn't live at the top-level of that repository, but in a subdirectory
+    called `cookiecutter`.
+
+    All information about your project will be prompted interactively by that
```
Comment (Marenz) [inline docs/index.md:55]: Very good

**Review by Marenz — APPROVED** [review]
One possible syntax problem, but lgtm

File: cookiecutter/variable-reference.md (hunk 2 above)
Comment (llucax) [reply cookiecutter/variable-reference.md:50]: Yes it is :star_struck:

**Pull request merged**